packages/angular_devkit/build_angular/src/browser/index.ts

```diff
@@ -231,50 +239,183 @@ export function buildWebpackBrowser(
           1,
         ),
         bufferCount(configs.length),
-        switchMap(buildEvents => {
+        switchMap(async buildEvents => {
+          configs.length = 0;
           const success = buildEvents.every(r => r.success);
-          if (success && options.index) {
+          if (success) {
             let noModuleFiles: EmittedFiles[] | undefined;
             let moduleFiles: EmittedFiles[] | undefined;
             let files: EmittedFiles[] | undefined;
 
-            const [firstBuild, secondBuild] = buildEvents;
-            if (isDifferentialLoadingNeeded) {
-              const scriptsEntryPointName = normalizeExtraEntryPoints(options.scripts || [], 'scripts')
-                .map(x => x.bundleName);
+            const scriptsEntryPointName = normalizeExtraEntryPoints(
+              options.scripts || [],
+              'scripts',
+            ).map(x => x.bundleName);
 
+            const [firstBuild, secondBuild] = buildEvents;
+            if (isDifferentialLoadingNeeded && (fullDifferential || options.watch)) {
               moduleFiles = firstBuild.emittedFiles || [];
-              files = moduleFiles.filter(x => x.extension === '.css' || (x.name && scriptsEntryPointName.includes(x.name)));
+              files = moduleFiles.filter(
+                x => x.extension === '.css' || (x.name && scriptsEntryPointName.includes(x.name)),
+              );
 
               if (buildEvents.length === 2) {
                 noModuleFiles = secondBuild.emittedFiles;
               }
+            } else if (isDifferentialLoadingNeeded && !fullDifferential) {
+              const { emittedFiles = [] } = firstBuild;
+              moduleFiles = [];
+              noModuleFiles = [];
+
+              // Common options for all bundle process actions
+              const actionOptions = {
+                optimize: normalizeOptimization(options.optimization).scripts,
+                sourceMaps:
+                  options.sourceMap === true ||
+                  (typeof options.sourceMap === 'object' && options.sourceMap.scripts),
+              };
+
+              const actions: {}[] = [];
+              const seen = new Set<string>();
+              for (const file of emittedFiles) {
+                // Scripts and non-javascript files are not processed
+                if (
+                  file.extension !== '.js' ||
+                  (file.name && scriptsEntryPointName.includes(file.name))
```

Comment:
This will cause an issue.

Scripts need to be processed or terser options in webpack need to be modified not to emit certain es6 optimizations.

One use case is when a variable in names the same as an object literal it will be emitted as such;

```js
var foo = true;
var bar = {foo:foo};
```

```js
var foo = true;
var bar = {foo};
```

Which will break in legacy browsers, can't quite remember the setting.

Reply:
Done.
The `ecma` option controls this.  It's now fixed to 5 for bundle differential builds.  The bundle processing will add the 6 optimizations to the ES2015 bundles.